pkg/translator/prometheusremotewrite/helper.go

```diff
@@ -258,7 +258,7 @@ func addSingleHistogramDataPoint(pt pmetric.HistogramDataPoint, resource pcommon
 		labels := make([]prompb.Label, len(baseLabels), len(baseLabels)+extraLabelCount+1) // +1 for name
 		copy(labels, baseLabels)
 
-		for extrasIdx := 0; extrasIdx < extraLabelCount; extrasIdx++ {
+		for extrasIdx := 0; extrasIdx < len(extras); extrasIdx += 2 {
 			labels = append(labels, prompb.Label{Name: extras[extrasIdx], Value: extras[extrasIdx+1]})
```

Comment:
Since `createLabels` is a nested function there isn't any easy way to directly test this, but would it make sense to have some kind of verification that `len(extras)` is even? This would make sure we don't have some kind of out of bounds error in a possible future usage, with `extras[extrasIds+1]`.

Reply:
I'll check.

Reply:
@crobert-1 since I realized that the two `createLabels` closures are practically the same, I refactored them into a single free function instead, that handles when `extras` is of uneven length. I also wrote a comprehensive test suite for `createLabels`. WDYT?